tests/integration/test_retry_producer.py

```diff
@@ -82,7 +80,7 @@ def test_retry_failed_payments(self):
         nb_success = len([row for row in success_report_rows if row.paid == PaymentStatus.PAID])
         nb_hash_xxx_op_hash = len([row for row in success_report_rows if row.hash == 'xxx_op_hash'])
 
-        self.assertEqual(31, nb_success)
+        self.assertEqual(26, nb_success)
```

Comment:
Just to make sure I understand it correctly:
There are total 31 batch payments. 5 are `nb_hash_xxx_op_hash` which are not successful thus making 26 sucessful payments right?

Reply:
@vkresch This is a test about retrying failed operations. Input is a past payment report which contains 31 payment items, 26 of them were successful and 5 of them were failed. 

The test checks that all items are successful in the final report, so there must be 31 successful items.
And also checks that only the previously failed items are repaid. This is crucial, already paid items must not be paid again, so only 5 of them have the test-specific hash value. 

Even though the test configuration was correct, I admit it is not readable.

Reply:
@habanoz shouldn't the final report have the `DONE` payment status for all 31 items instead of `PAID` and `INJECTED`?

Reply:
We have the following terminal states:
PAID: payment successfully made. 
FAIL: Some failures happened in the process.
DONE: Process completed without payment. E.g. zero amount, dry run...
INJECT: Transaction is injected into the node but after waiting for some time it is not added to any block. TRD does not know its fate. 

PAID and DONE means success. INJECT means unknown. 

In this particular test, we do not expect any payments with 'DONE' status.

Reply:
@habanoz  thanks for the clarification! I updated the unit test. See my comment further below.